week10/pages/Detail.jsx

```diff
@@ -0,0 +1,159 @@
+import { useState, useEffect } from "react";
+import { PostDetail, EditPost, DeletePost } from "../api/post";
+import { PutHeart, DeleteHeart, CntHeart } from "../api/heart";
+import { CreateComment, DeleteComment, GetComment } from "../api/comment";
+import styled from "styled-components";
+import { useParams, useNavigate } from "react-router-dom";
+const Title = styled.div``;
+const Content = styled.div``;
+const Image = styled.img`
+  height: 300px;
+  width: 300px;
+`;
+const CommentList = styled.div``;
+const CommentItem = styled.div``;
+const Detail = () => {
+  const { postId } = useParams();
+  const [post, setPost] = useState(null);
+  const [likeCnt, setLikeCnt] = useState(0);
+  const [comment, setComment] = useState();
+  const [comments, setComments] = useState([]);
+  const [commentCnt, setCommentCnt] = useState(0);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState("");
+  const [editedContent, setEditedContent] = useState("");

```

Comment:
P5: 상태관리 및 변수명을 약간 변경하면 가독성이 더 좋아질 것 같아요! 아래는 예시일 뿐이니 참고해보세요. ㅎㅎ

```suggestion
const [interaction, setInteraction] = useState({
  likeCount: 0,
  commentCount: 0,
});
const [editedPost, setEditedPost] = useState({ title: "", content: "" });
const [newComment, setNewComment] = useState("");
```